lunavl/sdk/estimators/body_estimators/body_attributes.py

```diff
@@ -429,6 +366,193 @@ def asDict(self) -> List[str]:
+                result.append(value)
+
+        if not result:
+            result.append(self._unknown)
+        return result
+
+
+class HeadwearState(BodyEstimation, ColorMixin):
+    """
+    Class for Headwear state estimation.
+
+    Headwear state properties:
+
+        - yes
+        - no
+        - unknown
+        - predominantState
+    """
+
+    _colors = {
+        "isBlack": "black",
+        "isWhite": "white",
+        "isOther": "other",
+        "isUnknown": "undefined",
+    }
+    _unknown = "undefined"
+
+    @property
+    def yes(self) -> float:
+        """
+        Get `yes` predict value.
+
+        Returns:
+            value in range [0, 1]
+        """
+        return self._coreEstimation.hat
+
+    @property
+    def no(self):
+        """
+        Get `no` predict value.
+
+        Returns:
+            value in range [0, 1]
+        """
+        return self._coreEstimation.noHat
+
+    @property
+    def unknown(self):
+        """
+        Get `unknown` predict value.
+
+        Returns:
+            value in range [0, 1]
+        """
+        return self._coreEstimation.unknown
+
+    @property
+    def predominantState(self) -> HeadwearStateEnum:
+        """Get headwear predominant state"""
+        return HeadwearStateEnum.fromCoreHeadwear(self._coreEstimation.result)
+
+    @property
+    def apparentColor(self):
+        return self.singleColor(self._coreEstimation.hatColor)
+
+    def asDict(self) -> dict:
+        """
+        Convert to dict.
+
+        Returns:
+            dict in platform format
+        """
+        return {
+            "apparent_color": self.apparentColor,
+            "predominant_state": str(self._coreEstimation.result.name).lower(),
+            "estimations": {
+                "yes": self.yes,
+                "no": self.no,
+                "unknown": self.unknown,
+            },
+        }
+
+
+class Shoes(BodyEstimation, ColorMixin):
+    """Shoes estimation."""
+
+    _colors = {
+        "isBlack": "black",
+        "isWhite": "white",
+        "isOther": "other",
+        "isUnknown": "undefined",
+    }
+    _unknown = "undefined"
+
+    @property
+    def apparentColor(self) -> str:
+        """Return name of the estimated shoe color."""
+        return self.singleColor(self._coreEstimation)
+
+    def asDict(self) -> dict:
+        """Serialize to dict."""
+        return {"apparent_color": self.apparentColor}
+
+
+class LowerGarment(BodyEstimation, ColorMixin):
+    """Lower garment estimantion."""
+
+    _colors = {
+        "isBlack": "black",
+        "isWhite": "white",
+        "isBlue": "blue",
+        "isGreen": "green",
+        "isGrey": "gray",
+        "isOrange": "orange",
+        "isPurple": "purple",
+        "isRed": "red",
+        "isYellow": "yellow",
+        "isPink": "pink",
+        "isBrown": "brown",
+        "isBeige": "beige",
+        "isKhaki": "khaki",
+        "isMulticolored": "multicolored",
```

Comment:
Researchers give a promise that will be used only two palettes.   `OutwearColorEnum` exists now already. Seems like it needs to rename it  and use for LowerGarment and OutwearColor. Also need create palette for HeadWear, shoes and other.

Reply:
unified